src/pmdas/openvswitch/pmdaopenvswitch.py

```diff
@@ -0,0 +1,411 @@
+                            self.port_info_metrics[item][0],
+                            pmdaMetric(self.pmid(self.port_info_cluster, item),
+                                       self.port_info_metrics[item][1],
+                                       self.port_info_indom,
+                                       self.port_info_metrics[item][2],
+                                       self.port_info_metrics[item][3]),
+                            self.port_info_metrics[item][4],
+                            self.port_info_metrics[item][4])
+
+        self.flow_indom = self.indom(2)
+        self.flow_instances()
+        self.flow_cluster = 2
+        self.flow_metrics = [
+            # Name - type - semantics - units - help
+            [ 'flow.cookie',                                                           PM_TYPE_STRING, PM_SEM_INSTANT, pmUnits(),  ''], # 0
+            [ 'flow.duration',                                                         PM_TYPE_FLOAT, PM_SEM_INSTANT, pmUnits(0,1,0,0,PM_TIME_SEC,0),  ''], # 1
+            [ 'flow.table',                                                            PM_TYPE_U64, PM_SEM_INSTANT, pmUnits(0,0,1,0,0,PM_COUNT_ONE),  ''], # 2
+            [ 'flow.n_packets',                                                        PM_TYPE_U64, PM_SEM_INSTANT, pmUnits(0,0,1,0,0,PM_COUNT_ONE),  ''], # 3
+            [ 'flow.n_bytes',                                                          PM_TYPE_U64, PM_SEM_INSTANT, pmUnits(0,0,1,0,0,PM_COUNT_ONE),  ''], # 4
+            [ 'flow.actions',                                                          PM_TYPE_STRING, PM_SEM_DISCRETE, pmUnits(),  ''], #5
+        ]
+
+        for item in range(len(self.flow_metrics)):
+            self.add_metric(name + '.' +
+                            self.flow_metrics[item][0],
+                            pmdaMetric(self.pmid(self.flow_cluster, item),
+                                       self.flow_metrics[item][1],
+                                       self.flow_indom,
+                                       self.flow_metrics[item][2],
+                                       self.flow_metrics[item][3]),
+                            self.flow_metrics[item][4],
+                            self.flow_metrics[item][4])
+
+
+        self.set_fetch_callback(self.openvswitch_fetch_callback)
+        self.set_refresh(self.openvswitch_refresh)
+
+    def fetch_switch_info(self):
+        """ fetches result from command line """
+        out = subprocess.Popen([os.path.join(PMDA_DIR, 'openvswitch', 'switch.sh')], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        stdout, stderr = out.communicate()
+        stdout = stdout.decode("utf-8")
+        return stdout, stderr
+
+    def get_switch_info_json(self):
+        """ Convert the commandline output to json """
+        stdout, _ = self.fetch_switch_info()
+
+        # Store it in a file
+        text_file = open("switch_output.txt", "w")
```

Comment:
I'm not understanding the need for this intermediate file - and its creation can lead to problems (e.g. ENOSPC) we don't want to have to deal with - ideally, just read directly from the ovs programs stdout.